### ISSUE TITLE
Clear Number on Country Change

### ISSUE THREAD

**Issue by wjoeyu** [catamphetamine/react-phone-number-input#299]
I want to achieve the workflow where when I select a new country, the current phone number value is cleared. This works as a temporary workaround for the issue presented here: https://github.com/catamphetamine/react-phone-number-input/issues/295

Right now, I try to pass a function to the onCountryChange prop that clears the value, but it always updates the value back to what the string was prior to changing the country. I see that the onChange function is called immediately after the onCountryChange function.

I was perusing through the docs, and I couldn't find a prop that was preventing me from clearing the value of the phone number. I am wondering if there was a way to achieve this.

**Comment by catamphetamine** [comment]
> Right now, I try to pass a function to the onCountryChange prop that clears the value, but it always updates the value back to what the string was prior to changing the country.

Perhaps that's a bug somewhere in the code.
I could look at it in some future.

> This works as a temporary workaround for the issue presented here: #295

Is #295 really such an issue?
The fix for it is quick.
Perhaps I should publish it already.
Maybe today or tomorrow.

**Comment by wjoeyu** [comment]
https://github.com/catamphetamine/react-phone-number-input/issues/295 is not this issue, but I'm trying to use the clearing on country change to kind of prevent the potential work flow in #295 from happening. Thanks so much!

**Comment by catamphetamine** [comment]
I investigated the issue and found the reason of the behavior in code:
https://github.com/catamphetamine/react-phone-number-input/blob/dc8d1676ae5b2f07f12781ca55e45b3bf6a684fb/source/PhoneInput.js#L606
When `onCountryChange()` is called, a new `value` is calculated, and then `onChange(newValue)` is called in something like `setTimeout()`, so it overrides `value` after you've reset it manually inside your `onCountryChange()`.
I've released a new version that fixed #295 : it now resets the phone number on country change if international number is input and the country calling codes of the two countries don't match.
See if this behavior fits your scenario.
